### PR TITLE
std: remove meta.Tuple, native tuple construction is here now

### PR DESCRIPTION
**Pull request by nektro**
not a specific issue but the feature has been around for a while now. cleaned up the last stragglers of its usage

### CODE REVIEW

File: lib/std/meta.zig [hunk 1]
```diff
@@ -990,17 +990,6 @@ pub fn ArgsTuple(comptime Function: type) type {
...
-/// with those types as fields.
-///
-/// Examples:
-/// - `Tuple(&[_]type {})` ⇒ `tuple { }`
-/// - `Tuple(&[_]type {f32})` ⇒ `tuple { f32 }`
-/// - `Tuple(&[_]type {f32,u32})` ⇒ `tuple { f32, u32 }`
-pub fn Tuple(comptime types: []const type) type {
-    return CreateUniqueTuple(types.len, types[0..types.len].*);
-}
-
```
Comment (thejoshwolfe) [inline lib/std/meta.zig:1003]: We could replace this with:

```zig
pub const Tuple = @compileError("Deprecated: Use struct { T1, T2 } syntax instead.");
```

but it seems pretty low stakes.

**Review by thejoshwolfe — APPROVED** [review]
I'm not sure what's up with the test failures, but this PR looks good from my perspective.

**Comment by InKryption** [comment]
I don't think `std.meta.Tuple` is entirely useless, it can actually be a relatively useful tool for generating generic argument tuples where the number of elements isn't known, and therefore couldn't be constructed using the in-built tuple syntax. Or even just for reducing verbosity e.g. `std.meta.Tuple(&.{type} ** 8)` vs `struct { type, type, type, type, type, type, type, type }`.
Though I'm not strongly inclined to fight its removal either, since this is a relatively niche use case and can still be replicated where needed.

**Comment by nektro** [comment]
the CI failure is from that `struct { foo }` and `@Type(Struct{ .fields = .{ .type= foo }})` return different intern pool indexes and thus different types

**Comment by nektro** [comment]
why wouldn't they be equivalent?

**Comment by Vexu** [comment]
Because the structural equivalency only applies to tuples created with tuple syntax, not to types created by `@Type` or the `struct` keyword.

**Comment by mlugg** [comment]
This behaviour is expected to change to give true structural equivalency for all tuple types - I am planning a refactor to tuples soon (this has previously been discussed in compiler meetings).

**Comment by andrewrk** [comment]
Closing abandoned pull request (CI checks failing, no activity in 30 days).

**Pull request closed without merging**